scripts/operations/configuration/apply_csm_configuration.sh

```diff
@@ -148,133 +206,145 @@ while [[ $# -gt 0 ]]; do
-fi
+    # If a file is passed in as input, keep a copy in the temporary directory
+    if [[ -n ${OLD_NCN_CONFIG_FILE} && -f ${OLD_NCN_CONFIG_FILE} ]]; then
+        run_cmd cp "${OLD_NCN_CONFIG_FILE}" "${TMPDIR}"
+    fi
+
+    run_mktemp --tmpdir="${TMPDIR}" "csm-config-$VERSION-XXXXXX.json"
+    CSM_CONFIG_FILE=${tmpfile}
+
+    run_mktemp --tmpdir="${TMPDIR}" "new-ncn-personalization-XXXXXX.json"
+    NCN_CONFIG_FILE=${tmpfile}
 
-if [[ -z "${COMMIT}" ]]; then
-    VCS_USER=$(kubectl get secret -n services vcs-user-credentials --template='{{.data.vcs_username}}' | base64 --decode)
-    [[ -n "${VCS_USER}" ]] || err_exit "Unable to obtain VCS username"
-    VCS_PASSWORD=$(kubectl get secret -n services vcs-user-credentials --template='{{.data.vcs_password}}' | base64 --decode)
-    [[ -n "${VCS_PASSWORD}" ]] || err_exit "Unable to obtain VCS password"
-    run_mktemp -d
-    TEMP_DIR=${tmpfile}
-    TEMP_HOME=${HOME}
-    HOME=${TEMP_DIR}
-    cd "${TEMP_DIR}" || err_exit "Unable to change directory to '${TEMP_DIR}'"
-    echo "${CLONE_URL/\/\//\/\/${VCS_USER}:${VCS_PASSWORD}@}" > .git-credentials
-    run_cmd git config --file .gitconfig credential.helper store
-    COMMIT=$(git ls-remote ${CLONE_URL} refs/heads/cray/csm/${VERSION} | awk '{print $1}')
-    if [[ -n ${COMMIT} ]]; then
-        echo "Found git commit ${COMMIT}"
+    run_mktemp --tmpdir="${TMPDIR}" "backup-ncn-personalization-XXXXXX.json"
+    BACKUP_NCN_CONFIG_FILE=${tmpfile}
+
+    if [[ -z "${CLONE_URL}" ]]; then
+        CLONE_URL="https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git"
+    fi
+
+    if [[ -z "${COMMIT}" ]]; then
+        VCS_USER=$(kubectl get secret -n services vcs-user-credentials --template='{{.data.vcs_username}}' | base64 --decode)
+        [[ -n "${VCS_USER}" ]] || err_exit "Unable to obtain VCS username"
+        VCS_PASSWORD=$(kubectl get secret -n services vcs-user-credentials --template='{{.data.vcs_password}}' | base64 --decode)
+        [[ -n "${VCS_PASSWORD}" ]] || err_exit "Unable to obtain VCS password"
+        run_mktemp -d
+        TEMP_DIR=${tmpfile}
+        TEMP_HOME=${HOME}
+        HOME=${TEMP_DIR}
+        cd "${TEMP_DIR}" || err_exit "Unable to change directory to '${TEMP_DIR}'"
+        echo "${CLONE_URL/\/\//\/\/${VCS_USER}:${VCS_PASSWORD}@}" > .git-credentials
+        run_cmd git config --file .gitconfig credential.helper store
+        COMMIT=$(git ls-remote ${CLONE_URL} refs/heads/cray/csm/${VERSION} | awk '{print $1}')
+        if [[ -n ${COMMIT} ]]; then
+            echo "Found git commit ${COMMIT}"
+        else
+            echo "No git commit found"
+            exit 1
+        fi
+        cd - >/dev/null || err_exit "Unable to change directory back to previous directory"
+        HOME=${TEMP_HOME}
+        rm -r "${TEMP_DIR}" || echo "WARNING: Unable to delete temporary directory '${TEMP_DIR}'" 1>&2
+    fi
+
+    #shellcheck disable=SC2089
```

Comment:
I don't think we need this anymore, now that the variable is set on a single line
```suggestion
```